tests/unit/grpc_internals/zeebe_job_adapter_test.py

```diff
@@ -25,184 +24,176 @@ def create_random_task_and_activate(grpc_servicer, task_type: str = None) -> str
-    job = get_first_active_job(task_type)
-    zeebe_job_adapter.fail_job(job_key=job.key, message=str(uuid4()))
+    job = get_first_active_job(task_type, zeebe_adapter)
+    zeebe_adapter.fail_job(job_key=job.key, message=str(uuid4()))
     with pytest.raises(JobAlreadyDeactivated):
-        zeebe_job_adapter.fail_job(job_key=job.key, message=str(uuid4()))
+        zeebe_adapter.fail_job(job_key=job.key, message=str(uuid4()))
 
 
-def test_fail_job_common_errors_called(grpc_servicer):
-    zeebe_job_adapter._common_zeebe_grpc_errors = MagicMock()
+def test_fail_job_common_errors_called(zeebe_adapter, grpc_servicer):
+    zeebe_adapter._common_zeebe_grpc_errors = MagicMock()
     error = grpc.RpcError()
     error._state = GRPCStatusCode(grpc.StatusCode.INTERNAL)
 
-    zeebe_job_adapter._gateway_stub.FailJob = MagicMock(side_effect=error)
+    zeebe_adapter._gateway_stub.FailJob = MagicMock(side_effect=error)
 
     task_type = create_random_task_and_activate(grpc_servicer)
-    job = get_first_active_job(task_type)
-    zeebe_job_adapter.fail_job(job_key=job.key, message=str(uuid4()))
+    job = get_first_active_job(task_type, zeebe_adapter)
+    zeebe_adapter.fail_job(job_key=job.key, message=str(uuid4()))
 
-    zeebe_job_adapter._common_zeebe_grpc_errors.assert_called()
+    zeebe_adapter._common_zeebe_grpc_errors.assert_called()
 
 
-def test_throw_error(grpc_servicer):
+def test_throw_error(zeebe_adapter, grpc_servicer):
     task_type = create_random_task_and_activate(grpc_servicer)
-    job = get_first_active_job(task_type)
-    response = zeebe_job_adapter.throw_error(job_key=job.key, message=str(uuid4()))
+    job = get_first_active_job(task_type, zeebe_adapter)
+    response = zeebe_adapter.throw_error(job_key=job.key, message=str(uuid4()))
     assert isinstance(response, ThrowErrorResponse)
 
 
-def test_throw_error_job_not_found():
+def test_throw_error_job_not_found(zeebe_adapter):
     with pytest.raises(JobNotFound):
-        zeebe_job_adapter.throw_error(job_key=randint(0, RANDOM_RANGE), message=str(uuid4()))
+        zeebe_adapter.throw_error(job_key=randint(0, RANDOM_RANGE), message=str(uuid4()))
 
 
-def test_throw_error_already_thrown(grpc_servicer):
+def test_throw_error_already_thrown(zeebe_adapter, grpc_servicer):
     task_type = create_random_task_and_activate(grpc_servicer)
-    job = get_first_active_job(task_type)
-    zeebe_job_adapter.throw_error(job_key=job.key, message=str(uuid4()))
+    job = get_first_active_job(task_type, zeebe_adapter)
+    zeebe_adapter.throw_error(job_key=job.key, message=str(uuid4()))
     with pytest.raises(JobAlreadyDeactivated):
-        zeebe_job_adapter.throw_error(job_key=job.key, message=str(uuid4()))
+        zeebe_adapter.throw_error(job_key=job.key, message=str(uuid4()))
 
 
-def test_throw_error_common_errors_called(grpc_servicer):
-    zeebe_job_adapter._common_zeebe_grpc_errors = MagicMock()
+def test_throw_error_common_errors_called(zeebe_adapter, grpc_servicer):
```

Comment:
This is defined twice (not introduced now - and renaming it will result in the first failing)